chore: migrate from jest to vitest

luipe opened the pull request:
### Test Execution Times (locally on my machine)

- for this branch: 73.91s
- for main: 117.541s

weissu42 committed:
test: remove duplicate test, use shorter intervals to shorten test execution by over 20% on my machine

Signed-off-by: Ute Weiß <ute.weiss@tngtech.com>
Signed-off-by: Luisa Peter <luisa.peter@tngtech.com>

luipe committed:
chore: migrate from jest to vitest

Signed-off-by: Luisa Peter <luisa.peter@tngtech.com>

NiklasEi commented:
This is a good idea, I much prefer vitest over jest. :+1: 
Then we can also finally get rid of #957 and #963.

luipe committed:
test: make times in tests more traceable

Signed-off-by: Luisa Peter <luisa.peter@tngtech.com>

luipe commented:
@weissu42 I included your commit for test improvements in this PR.

luipe committed:
test: fix flaky test

"long running interval jobs does not start a long running job twice that should not run in parallel" had a race condition due to the duration being a multiple of the interval.

Signed-off-by: Luisa Peter <luisa.peter@tngtech.com>

luipe commented:
Adresses #937
Final result for running all tests on my machine is 48s. A flaky test was fixed as well.

NiklasEi reviewed (approved):
Looks good, thank you!
